cmd/goal/node.go

```diff
@@ -374,3 +390,89 @@ var waitCmd = &cobra.Command{
 		}
 	},
 }
+
+var createCmd = &cobra.Command{
+	Use:   "create",
+	Short: "create a node at the desired data directory for the desired network",
+	Long:  "create a node at the desired data directory for the desired network",
+	Args:  validateNoPosArgsFn,
+	Run: func(cmd *cobra.Command, _ []string) {
+
+		validNetworks := map[string]bool{"mainnet": true, "testnet": true, "devnet": true}
+		if !validNetworks[newNodeNetwork] {
+			reportErrorf(errorNodeCreation, "passed network name invalid")
+		}
+
+		gopath := os.Getenv("GOPATH")
+		if gopath == "" {
+			gopath = build.Default.GOPATH
+		}
+		firstChoicePath := filepath.Join(gopath, "bin", "genesisfiles", newNodeNetwork, "genesis.json")
+		secondChoicePath := filepath.Join("var", "lib", "algorand", "genesis", newNodeNetwork, "genesis.json")
+		thirdChoicePath := filepath.Join(gopath, "bin", "genesisfiles", "genesis", newNodeNetwork, "genesis.json")
+		paths := [...]string{firstChoicePath, secondChoicePath, thirdChoicePath}
+		correctPath := ""
+		for _, pathCandidate := range paths {
+			if _, err := os.Stat(pathCandidate); err == nil {
+				correctPath = pathCandidate
+				break
+			}
+		}
+		if correctPath == "" {
+			reportErrorf(errorNodeCreation, "no genesis file found")
+		}
+		genesisInput, err := ioutil.ReadFile(correctPath)
+		if err != nil {
+			reportErrorf(errorNodeCreation, err)
+		}
+		destPath := filepath.Join(newNodeDestination, "genesis.json")
+		err = ioutil.WriteFile(destPath, genesisInput, 0666)
+		if err != nil {
+			reportErrorf(errorNodeCreation, err)
+		}
+
+		var configBuilder strings.Builder
```

Comment:
Populate a Local config struct from the inputs and call "codec.SaveNonDefaultValuesToFile(filename, config, config.GetDefaultLocal(), nil, true)